xbmc/video/VideoDatabase.cpp

```diff
@@ -5875,11 +5873,9 @@ bool CVideoDatabase::GetSeasonsByWhere(const std::string& strBaseDir, const Filt
       int showId = m_pDS->fv(VIDEODB_ID_SEASON_TVSHOW_ID).get_asInt();
       int iSeason = m_pDS->fv(VIDEODB_ID_SEASON_NUMBER).get_asInt();
       std::string name = m_pDS->fv(VIDEODB_ID_SEASON_NAME).get_asString();
-      std::string path = m_pDS->fv(VIDEODB_ID_SEASON_TVSHOW_PATH).get_asString();
 
       if (mapSeasons.find(std::make_pair(showId, iSeason)) == mapSeasons.end() &&
-         (CProfilesManager::GetInstance().GetMasterProfile().getLockMode() == LOCK_MODE_EVERYONE || g_passwordManager.bMasterUser ||
-          g_passwordManager.IsDatabasePathUnlocked(path, *CMediaSourceSettings::GetInstance().GetSources("video"))))
+         (CProfilesManager::GetInstance().GetMasterProfile().getLockMode() == LOCK_MODE_EVERYONE || g_passwordManager.bMasterUser))
```

Comment:
I actually didn't understand this at first. This checks if the path is one of the sources and in case it is it will add them even if it's locked.
My change is wrong.. If I do this in a locked account it won't show any season.. So probably we can't simply drop strpath afterall.
But shouldn't this check belong in tvshow too? It doesn't make sense we show every tvshow without checking and do the check when we load the seasons.